api.go

```diff
@@ -73,6 +67,52 @@ func (s *Session) apiRequestWithParams(endpoint string, mixins []string, params
 	return response.Payload, nil
 }
 
+// mockRequester answers API requests by returning some stock response.
+type mockRequester struct {
+	message *json.RawMessage
+}
+
+// request pretends to fulfil an API request, but actually returns the mockRequester's stock response.
+func (s *mockRequester) request(endpoint string, mixins []string, params map[string][]string) (*json.RawMessage, error) {
+	return s.message, nil
+}
+
+// Session represents an open API session.
+type Session struct {
+	requester apiRequester
+}
+
+// NewSession constructs a new Session with the given API key.
+func NewSession(apikey string) (*Session, error) {
+	url, err := url.Parse(`https://ury.york.ac.uk/api/v2`)
```

Comment:
wait, why was this ury.york.ac.uk in the original? Also, why isn't configurable?